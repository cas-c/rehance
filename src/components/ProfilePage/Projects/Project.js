import React from 'react';
import PropTypes from 'prop-types';

const Project = ({ data }) => (
    <a href={data.url} target='_blank' title={data.name} >
        <div className='card'>
            <header className='card-header'>
                <p className='card-header-title'>
                    { data.name }
                </p>
            </header>
            <div className='card-image project-cover'>
                <figure className='image is-4by3'>
                    {
                        data.covers && <img className='project-cover-img' src={data.covers.original} alt={data.name} />
                    }
                </figure>
            </div>
        </div>
    </a>
);

Project.propTypes = {
    data: PropTypes.shape({
        url: PropTypes.string,
        name: PropTypes.string,
        covers: PropTypes.shape({
            original: PropTypes.string
        })
    })
}

Project.defaultProps = {
    data: {}
}

export default Project;
