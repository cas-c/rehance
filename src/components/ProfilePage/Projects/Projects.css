/* todo: handle longer project names */


.project-cover {
    overflow-y: hidden;
    align-content: center;
}

.project-cover-img {
    max-height: 15em;
    height: 100%;
    width: auto;
}