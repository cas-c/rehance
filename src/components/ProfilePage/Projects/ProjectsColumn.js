import React from 'react';
import PropTypes from 'prop-types';
import Project from './Project';

//todo: Can probably make 'Column' a more generic component (see MiniProfilesColumn)

const ProjectsColumn = ({ projects }) => (
    <div>
        <div className='columns'>
            {
                projects.length > 0 ?
                    projects.slice(0, 5).map((p, i) =>
                        <div className='column' key={`project-${i}-wrapper`}>
                            <Project data={p} key={`project-${i}`}/>
                        </div>
                    ) :
                    <div className='has-text-centered'> This user does not have any projects. </div>
            }
        </div>
        { projects.length > 5 && <div className='has-text-right'><a>View all...</a></div> }
    </div>
);

ProjectsColumn.propTypes = {
    projects: PropTypes.arrayOf(PropTypes.object)
};

ProjectsColumn.defaultProps = {
    projects: []
};

export default ProjectsColumn;
