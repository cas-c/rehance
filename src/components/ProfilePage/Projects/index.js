import React from 'react';
import ProjectsColumn from './ProjectsColumn';
import Project from './Project';
import './Projects.css';

class Projects extends React.Component {
    componentDidUpdate(prevProps) {
        if (this.props.projects[0] === undefined) return;
        if (prevProps.projects[0] === undefined || prevProps.projects[0].covers.original !== this.props.projects[0].covers.original) {
            this.props.setBackground(this.props.projects[0].covers.original);
        }
    }
    render() {
        return (
            <div>
                <div className='profile-page-header'>
                    <strong className='is-size-1'>Projects</strong>
                </div>
                <ProjectsColumn projects={this.props.projects} />
            </div>
        );
    }
}

Projects.defaultProps = {
    projects: []
}

export default Projects;
