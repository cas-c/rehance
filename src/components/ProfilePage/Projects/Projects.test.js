import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Projects from './index';

const mockProjects = [{
    url: 'Mock URL',
    name: 'Test Project Name',
    covers: {
        original: 'Original cover URL'
    }
}];

describe('Projects', () => {
    it('snapshot: should render correctly', () => {
        const output = shallow(<Projects projects={mockProjects} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    //todo: Test the background image switching function in componentDidUpdate
});
