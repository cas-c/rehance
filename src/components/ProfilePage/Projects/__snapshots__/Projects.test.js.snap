// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects snapshot: should render correctly 1`] = `
<div>
  <div
    className="profile-page-header"
  >
    <strong
      className="is-size-1"
    >
      Projects
    </strong>
  </div>
  <ProjectsColumn
    projects={
      Array [
        Object {
          "covers": Object {
            "original": "Original cover URL",
          },
          "name": "Test Project Name",
          "url": "Mock URL",
        },
      ]
    }
  />
</div>
`;
